Add documentation for here_travel_time (#9648)
home-assistant/home-assistant.github.io: source/_components/here_travel_time.markdown

---
layout: page
title: "HERE Travel Time"
description: "Instructions on how to add HERE travel time to Home Assistant."
date: 2019-06-06 15:06
sidebar: true
comments: false
sharing: true
footer: true
logo: HERE_logo.svg

> ha_category:

Reviewer: ```suggestion
ha_category:
  - Transport
  - Sensor
```

  - Transport
ha_iot_class: Cloud Polling
ha_release: 0.95

> redirect_from:

Reviewer 2: This is a new component, so redirects are not needed. Please remove this.

> redirect_from:

Reviewer: Please remove the `redirect_from`

 - /components/sensor.here_travel_time/
---

The `here_travel_time` sensor provides travel time from the [HERE Routing API](https://developer.here.com/documentation/routing/topics/introduction.html).

## {% linkable_title Setup %}

You need to register for an API key by following the instructions [here](https://developer.here.com/documentation/routing/topics/introduction.html?create=Freemium-Basic&keepState=true&step=account).

> HERE offers a Freemium Plan which includes 250.000 free Transactions per month. For the Routing API one transaction equals one request with one starting point (no multistop). More information can be found [here](https://developer.here.com/faqs#payment-subscription)

Reviewer 2: ```suggestion
HERE offers a Freemium Plan which includes 250.000 free Transactions per month. For the Routing API, one transaction equals one request with one starting point (no multistop). More information can be found [here](https://developer.here.com/faqs#payment-subscription)
```

By default HERE will deactivate your account if you exceed the free Transaction limit for the month. You can add payment details to reenable your account as described [here](https://developer.here.com/faqs)

## {% linkable_title Configuration %}

To enable the sensor, add the following lines to your `configuration.yaml` file:

```yaml
# Example entry for configuration.yaml

> sensor:

Reviewer: Only required items in the first example

Reviewer 2: This is the first example....

  - platform: here_travel_time
    app_id: "YOUR_APP_ID"
    app_code: "YOUR_APP_CODE"
    origin: "51.222975,9.267577"
    destination: "51.257430,9.335892"

>     name: "HERE Travel Time Home -> Work"

Reviewer 2: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

>     travel_mode: "car"

Reviewer 2: same

>     route_mode: "shortest"

Reviewer 2: same

>     traffic_mode: "true"

Reviewer 2: same

```

{% configuration %}
app_id:
  description: "Your application's API id (get one by following the instructions above)."
  required: true
  type: string
app_code:
  description: "Your application's API code (get one by following the instructions above)."
  required: true
  type: string
origin:
  description: "The starting point for calculating travel distance and time."
  required: true
  type: string
destination:
  description: "One or more locations to use as the finishing point for calculating travel distance and time."
  required: true
  type: string
name:
  description: A name to display on the sensor. The default is "HERE Travel Time".
  required: false
  type: string

> travel_mode:

Reviewer 2: According to the parent PR, this configuration option has a default but is not listed in the documentation. Please add the default.

  description: "You can choose between: `car`, `pedestrian`, `publicTransport` or `truck`. The default is `car`."
  required: false
  type: string

> route_mode:

Reviewer 2: According to the parent PR, this configuration option has a default but is not listed in the documentation. Please add the default.

  description: "You can choose between: `fastest`, or `shortest`. This will determine whether the route is optimized to be the shortest and completely disregard traffic and speed limits or the fastest route according to the current traffic information. The default is `fastest`"
  required: false
  type: string

> traffic_mode:

Reviewer 2: According to the parent PR, this configuration option has a default but is not listed in the documentation. Please add the default.

  description: "You can choose between: `true`, or `false`. Decide whether you want to take the current traffic condition into account. Default is `false`."
  required: false

>   type: string

Reviewer 2: According to the parent PR, this should be a boolean.

{% endconfiguration %}

## {% linkable_title Dynamic Configuration %}

> Tracking can be setup to track entities of type `device_tracker`, `zone`, `sensor` and `person`. If an entity is placed in the origin or destination then every 5 minutes when the platform updates it will use the latest location of that entity.

Reviewer 2: ```suggestion
Tracking can be set up to track entities of type `device_tracker`, `zone`, `sensor` and `person`. If an entity is placed in the origin or destination then every 5 minutes when the platform updates it will use the latest location of that entity.
```

```yaml
# Example entry for configuration.yaml
sensor:
  # Tracking entity to entity
  - platform: here_travel_time
    app_id: "YOUR_APP_ID"
    app_code: "YOUR_APP_CODE"
    name: Phone To Home
    origin: device_tracker.mobile_phone
    destination: zone.home

  # Tracking entity to zone friendly name
  - platform: here_travel_time
    app_id: "YOUR_APP_ID"
    app_code: "YOUR_APP_CODE"
    name: Home To Eddie's House
    origin: zone.home
    destination: Eddies House    # Friendly name of a zone
```

## {% linkable_title Entity Tracking %}

- **device_tracker**

>   - If state is a zone then the zone location will be used

Reviewer 2: ```suggestion
  - If the state is a zone, then the zone location will be used
```

>   - If state is not a zone it will look for the longitude and latitude attributes

Reviewer 2: ```suggestion
  - If the state is not a zone, it will look for the longitude and latitude attributes
```

- **zone**
  - Uses the longitude and latitude attributes
  - Can also be referenced by just the zone's friendly name found in the attributes.
- **sensor**

>   - If state is a zone or zone friendly name then will use the zone location

Reviewer 2: ```suggestion
  - If the state is a zone or zone friendly name then will use the zone location
```

  - All other states will be passed directly into the HERE API
    - This includes all valid locations listed in the *Configuration Variables*

## {% linkable_title Updating sensors on-demand using Automation %}

You can also use the `homeassistant.update_entity` service to update the sensor on-demand. For example, if you want to update `sensor.morning_commute` every 2 minutes on weekday mornings, you can use the following automation:

```yaml

> - id: update_morning_commute_sensor

Reviewer 2: You might want to start this with `automation:` to indicate more clearly that this is an automation.

---
title: "HERE Travel Time"
description: "Instructions on how to add HERE travel time to Home Assistant."
logo: HERE_logo.svg
ha_category:
  - Transport
  - Sensor
ha_iot_class: Cloud Polling

> ha_release: 0.100

Reviewer: ```suggestion
ha_release: "0.100"
```
